src/main/java/com/gradle/CiUtils.java

```diff
@@ -0,0 +1,71 @@
+package com.gradle;
+
+import org.gradle.api.provider.ProviderFactory;
+
+public class CiUtils {
```

Comment:
Two things,

I think we should make thing change separately and apply it to both the Maven extension and Gradle plugin for easier maintenance.


What do you think of a change where this class is instantiated to avoid passing a `ProviderFactory` to each method?

Reply:
We could do that, let me split the PR

Reply:
This PR is now addressing the issue only

Reply:
Ah sorry, I meant to start with a PR that refactors the checks for CI environment variables. This PR still contains some refactoring for CI environment variable checks. This way we can review the logic changes independently of the refactoring changes and we can apply that to the GE maven extension as well.